design/Events_design.md

```diff
@@ -0,0 +1,113 @@
+# Events in Obscuro - Design document
+
+## Scope
+
+This is a design proposal for how to handle events in Obscuro.
+
+## What are events in Ethereum
+
+### Smart contracts can emit events which are streamed from a node and generally used by UIs.
+
+For example:
+
+```emit Transfer(from, to, amount);```
+
+
+### A web app can subscribe to events by doing something like:
+
+```
+var subscription = web3.eth.subscribe('logs', {
+    address: '0x123456..', 
+    topics: ['0x12345...']
+}, function(error, result){
+    if (!error)
+        console.log(result);
+});
+- address - String|Array: An address or a list of addresses to only get logs from particular account(s).
+- topics - Array: An array of values which must each appear in the log entries. The order is important, if you want to leave topics out use null, e.g. [null, '0x00...']. You can also pass another array for each topic with options for that topic e.g. [null, ['option1', 'option2']]
```

Comment:
Can we add a bit to this doc explaining how topics are set? It's not clear to me, and I think it's necessary info for understanding the design. There seems to be some implication that a topic is automatically set based on the `from` and/or `to` fields, perhaps?